### ISSUE TITLE
 Invalid platform header count in IGVM file

### ISSUE THREAD

**Issue by ramagali24** [coconut-svsm/svsm#375]
Any suggestions:

This is command i used  QEMU to launch an  SEV-SNP protected guest with the COCONUT-SVSM

sudo $HOME/bin/qemu-svsm/bin/qemu-system-x86_64 \
  -enable-kvm \
  -cpu EPYC-v4 \
  -machine q35,confidential-guest-support=sev0,memory-backend=ram1 \
  -object memory-backend-memfd,id=ram1,size=8G,share=true,prealloc=false,reserve=false \
  -object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,igvm-file=/home/amd/edk2/svsm/bin/coconut-qemu.igvm \
  -smp 8 \
  -no-reboot \
  -netdev user,id=vmnic -device e1000,netdev=vmnic,romfile= \
  -drive file=/home/amd/snp/launch/snp-guest.img,format=qcow2, \
  -device virtio-scsi-pci,id=scsi0,disable-legacy=on,iommu_platform=on \
  -device scsi-hd,drive=disk0,bootindex=0 \
  -vga std \
  -serial stdio \
  -serial pty

Error:
qemu-system-x86_64: Invalid platform header count in IGVM file. Error code: FFFFFFFF

**Comment by stefano-garzarella** [comment]
Can you try adding `init-flags=5` in the `sev-snp-guest` object?

BTW we have a script committed that you can use as reference: https://github.com/coconut-svsm/svsm/blob/main/scripts/launch_guest.sh

**Comment by ramagali24** [comment]
Thanks Stefano. This resolved current error and i am seeing now below one.

qemu-system-x86_64: -object memory-backend-memfd-private,size=8G,id=mem0,share=true: Parameter 'qom-type' does not accept value 'memory-backend-memfd-private

**Comment by stefano-garzarella** [comment]
> Thanks Stefano. This resolved current error and i am seeing now below one.

Great!

> qemu-system-x86_64: -object memory-backend-memfd-private,size=8G,id=mem0,share=true: Parameter 'qom-type' does not accept value 'memory-backend-memfd-private

mmm, I don't see `memory-backend-memfd-private` object in the cmdline you posted, did you change it?

BTW I suggest taking a look at https://github.com/coconut-svsm/svsm/blob/main/scripts/launch_guest.sh since depending on the QEMU version (from Coconut repo), we have different memory backend object to use:

```
# The QEMU machine and memory command line changed after QEMU 8.2.0 from
# the coconut-svsm git repository.
if (( (QEMU_MAJOR > 8) || ((QEMU_MAJOR == 8) && (QEMU_MINOR >= 2)) )); then
  MACHINE=q35,confidential-guest-support=sev0,memory-backend=mem0
  MEMORY=memory-backend-memfd,size=8G,id=mem0,share=true,prealloc=false,reserve=false
else
  MACHINE=q35,confidential-guest-support=sev0,memory-backend=mem0,kvm-type=protected
  MEMORY=memory-backend-memfd-private,size=8G,id=mem0,share=true
fi

```

**Comment by ramagali24** [comment]
I am able to clear above errors by setting environment variable to QEMU. 

And then i started getting below erros and to fix this i had to copy these files to the folder i am running script..

Could not open option rom 'kvmvapic.bin': No such file or directory
qemu-system-x86_64: failed to find romfile "vgabios-stdvga.bin

And then when i try launch guest, i see again below error. 

qemu-system-x86_64: IGVM file does not describe a compatible supported platform

Any suggestions, why i keep getting this.

**Comment by ramagali24** [comment]
This issue resolved after i moved to IGVM 0.1.6 version.

**Comment by stefano-garzarella** [comment]
> This issue resolved after i moved to IGVM 0.1.6 version.

Okay, so please close the issue if it's resolved.
